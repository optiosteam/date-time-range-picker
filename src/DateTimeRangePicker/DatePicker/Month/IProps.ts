import {Moment} from 'moment'
import {IData} from '../interfaces/IData'

interface IProps {
    month: number,
    year: number,
    fromDate?: Moment,
    untilDate?: Moment,
    hoverDate?: Moment,
    showDaysPreviousMonth: boolean,
    showDaysNextMonth: boolean,
    onDaySelected: (date: Moment) => void,
    onDayHover: (date?: Moment | undefined) => void
    data?: IData[]
}

export default IProps
