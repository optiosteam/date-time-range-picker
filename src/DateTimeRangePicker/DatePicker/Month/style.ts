import styled from 'styled-components'

const colorHighlight = '#428be3'
const colorBackgroundDay = 'rgba(black, .1)'
const colorOtherMonth = 'rgba(black, .4)'
const colorDisabled = 'rgba(black, .4)'
const colorAlreadySelected = 'rgba(red, .4)'


export const CalendarButton = styled.button<{
    active?: boolean,
    alreadySelected?: string,
    activeFromDate?: boolean,
    activeUntilDate?: boolean,
    otherMonth?: boolean,
    activeFromDateReverse?: boolean,
    inRange?: boolean,
    hover?: boolean,
    hoverPast?: boolean,
    hoverFuture?: boolean
}>`
  width: 2.1em;
  height: 2.1em;
  background-color: ${colorBackgroundDay};
  border-radius: 1.5em;
  margin: .3em;
  cursor: pointer;
  box-sizing: content-box;
  color: ${({otherMonth}) => otherMonth ? colorOtherMonth : 'inherit'};
  background-color: ${({active, alreadySelected}) => active
    ? (alreadySelected ? alreadySelected : colorHighlight)
    : (alreadySelected ? alreadySelected : 'inherit')};
  &:hover:not(.disabled),
  &.hover:not(.disabled) {
    background-color: rgba(${colorHighlight}, .8);
    color: white;
  }


`

export const CalendarButton2 = styled.button`
        > button {
          width: 2.1em;
          height: 2.1em;
          background-color: ${colorBackgroundDay};
          border-radius: 1.5em;
          margin: .3em;
          cursor: pointer;
          box-sizing: content-box;

          &:hover:not(.disabled),
          &.hover:not(.disabled) {
            background-color: rgba(${colorHighlight}, .8);
            color: white;
          }

          &.other-month {
            color: ${colorOtherMonth};
          }

          &.active {
            background-color: $color-highlight;
            color: white;
          }

          &.already-selected {
            background-color: $color-already-selected;
            margin: .3em 0;
            padding: 0 .3em;
            border-radius: 0;

            &.active-from-date:not(.active-from-date-reverse) {
              border-top-left-radius: 1.5em;
              border-bottom-left-radius: 1.5em;
              margin-left: .3em;
              padding-left: 0;
            }

            &.active-until-date {
              border-top-right-radius: 1.5em;
              border-bottom-right-radius: 1.5em;
              margin-right: .3em;
              padding-right: 0;
            }

            &.active {
              background-color: $color-already-selected;
            }
          }

          &.in-range {
            background-color: rgba($color-highlight, .5);
            margin: .3em 0;
            padding: 0 .3em;
            border-radius: 0;

            &.active-from-date:not(.active-from-date-reverse) {
              border-top-left-radius: 1.5em;
              border-bottom-left-radius: 1.5em;
              margin-left: .3em;
              padding-left: 0;
            }

            &.active-until-date {
              border-top-right-radius: 1.5em;
              border-bottom-right-radius: 1.5em;
              margin-right: .3em;
              padding-right: 0;
            }

            &.active {
              background-color: $color-highlight;
            }
          }

          &.hover-range:not(.in-range) {
            background-color: rgba($color-highlight, .5);
            margin: .3em 0;
            padding: 0 .3em;
            border-radius: 0;

            &.active-from-date:not(.active-from-date-reverse) {
              border-top-left-radius: 1.5em;
              border-bottom-left-radius: 1.5em;
              margin-left: .3em;
              padding-left: 0;

              &.hover {
                border-radius: 1.5em;
                padding-right: 0;
                margin-right: .3em;
              }
            }

            &.active-from-date-reverse {
              border-top-right-radius: 1.5em;
              border-bottom-right-radius: 1.5em;
              margin-right: .3em;
              padding-right: 0;
            }

            &.hover-past {
              border-top-left-radius: 1.5em;
              border-bottom-left-radius: 1.5em;
              padding-left: 0;
              margin-left: .3em;
            }

            &.hover-future {
              border-top-right-radius: 1.5em;
              border-bottom-right-radius: 1.5em;
              padding-right: 0;
              margin-right: .3em;
            }

            &.active {
              background-color: $color-highlight;
            }
          }

          &.disabled {
            color: $color-disabled;
            cursor: default;
          }
        }

`
