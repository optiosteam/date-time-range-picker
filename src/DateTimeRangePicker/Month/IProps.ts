import moment from 'moment'

interface IProps {
    date: moment.Moment,
    showDaysPreviousMonth: boolean,
    showDaysNextMonth: boolean
}

export default IProps
