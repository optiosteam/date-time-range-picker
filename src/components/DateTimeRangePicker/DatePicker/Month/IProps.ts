import {Moment} from 'moment'

interface IProps {
    month: number,
    year: number,
    fromDate?: Moment,
    untilDate?: Moment,
    hoverDate?: Moment,
    showDaysPreviousMonth: boolean,
    showDaysNextMonth: boolean,
    onDaySelected: (date: Moment) => void,
    onDayHover: (date?: Moment | undefined) => void
}

export default IProps
