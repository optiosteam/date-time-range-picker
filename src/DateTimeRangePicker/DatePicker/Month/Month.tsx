import React, {ReactElement, useState} from 'react'
import moment, {Moment} from 'moment'

import IProps from './IProps'

const Month: React.FunctionComponent<IProps> = (
    {
        displayRanges,
        month,
        year,
        fromDate,
        untilDate,
        hoverDate,
        showDaysPreviousMonth ,
        showDaysNextMonth,
        onDaySelected,
        onDayHover
    }
) => {
    const date = moment(`${month}-${year}`, 'M-YYYY')

    const getDays = (): ReactElement[] => {
        const currentDate = date.clone().startOf('month').startOf('week')
        const endDate = date.clone().endOf('month').endOf('week')
        const dates: Moment[] = []
        while (currentDate.isBefore(endDate)) {
            dates.push(currentDate.clone())
            currentDate.add(1, 'day')
        }

        return dates.map((buttonDate: Moment) => {
            const classNames = []

            displayRanges.forEach(displayRange => {
                displayRange.forEach(date => {
                    if (date.isSame(buttonDate, 'day')) {
                        classNames.push('already-selected')
                        if (displayRange[0].isSame(date)) {
                            classNames.push('active-from-date')

                        } else if (displayRange[displayRange.length - 1].isSame(date)) {
                            classNames.push('active-until-date')
                        }
                    }
                })
            })

            if (!buttonDate.isSame(date, 'month')) {
                if (buttonDate.isBefore(date) && !showDaysPreviousMonth) {
                    return <span className={'dummy-day'} key={`day${buttonDate.format('YYYYMMDD')}`}/>
                }

                if (buttonDate.isAfter(date) && !showDaysNextMonth) {
                    return <span className={'dummy-day'} key={`day${buttonDate.format('YYYYMMDD')}`}/>
                }

                classNames.push('other-month')
            }

            if (fromDate && buttonDate.isSame(fromDate, 'day')) {
                classNames.push('active')
                classNames.push('active-from-date')

                if (!untilDate && hoverDate && hoverDate.isBefore(buttonDate, 'day')) {
                    classNames.push('active-from-date-reverse')
                }
            }

            if (untilDate && buttonDate.isSame(untilDate, 'day')) {
                classNames.push('active')
                classNames.push('active-until-date')
            }

            if (fromDate && untilDate && buttonDate.isBetween(fromDate, untilDate, 'day', '[]')) {
                classNames.push('in-range')
            }

            if (
                fromDate
                && ! untilDate
                && hoverDate
                && (
                    buttonDate.isBetween(fromDate, hoverDate, 'day', '[]')
                    || buttonDate.isBetween(hoverDate, fromDate, 'day', '[]')
                )
            ) {
                classNames.push('hover-range')
            }

            if (hoverDate  && buttonDate.isSame(hoverDate, 'day')) {
                classNames.push('hover')

                if (fromDate && !untilDate) {
                    if (fromDate.isAfter(buttonDate, 'day')) {
                        classNames.push('hover-past')
                    }

                    if (fromDate.isBefore(buttonDate, 'day')) {
                        classNames.push('hover-future')
                    }
                }
            }

            return <button
                key={`day${buttonDate.format('YYYYMMDD')}`}
                className={classNames.join(' ')}
                onClick={
                    () => {
                        onDaySelected(buttonDate.clone())
                    }
                }
                onMouseEnter={
                    () => {
                        onDayHover(buttonDate.clone())
                    }
                }
                type={'button'}
            >
                {buttonDate.format('D')}
            </button>
        })
    }

    return <div className={'date-time-range-picker-month'}>
        <div className={'date-time-range-picker-month-daynames'}>
            <div>{date.startOf('week').format('dd')}</div>
            <div>{date.startOf('week').add(1, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(2, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(3, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(4, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(5, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(6, 'days').format('dd')}</div>
        </div>
        <div className={'date-time-range-picker-month-days'} onMouseLeave={() => onDayHover(undefined)}>
            {getDays()}
        </div>
    </div>
}

export default Month
