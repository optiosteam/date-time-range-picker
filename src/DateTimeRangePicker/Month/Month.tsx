import React, {ReactElement} from 'react'
import moment from 'moment'

import IProps from './IProps'

const Month: React.FunctionComponent<IProps> = (
    {
        date,
        showDaysPreviousMonth ,
        showDaysNextMonth
    }
) => {
    return <div className={'date-time-range-picker-month'}>
        <div className={'date-time-range-picker-month-daynames'}>
            <div>{date.startOf('week').format('dd')}</div>
            <div>{date.startOf('week').add(1, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(2, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(3, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(4, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(5, 'days').format('dd')}</div>
            <div>{date.startOf('week').add(6, 'days').format('dd')}</div>
        </div>
        <div className={'date-time-range-picker-month-days'}>
            {getDays(date, showDaysPreviousMonth, showDaysNextMonth)}
        </div>
    </div>
}

const getDays = (date: moment.Moment, showDaysPreviousMonth: boolean, showDaysNextMonth: boolean): ReactElement[] => {
    const days = []
    const currentDate = date.clone().startOf('month').startOf('week')
    const endDate = date.clone().endOf('month').endOf('week')
    while (currentDate.isBefore(endDate)) {
        const classNames = []
        if (! currentDate.isSame(date, 'month')) {
            if (currentDate.isBefore(date) && ! showDaysPreviousMonth) {
                days.push(<span className={'dummy-day'}/>)
                currentDate.add(1, 'day')
                continue
            }
            if (currentDate.isAfter(date) && ! showDaysNextMonth) {
                days.push(<span className={'dummy-day'}/>)
                currentDate.add(1, 'day')
                continue
            }

            classNames.push('other-month')
        }

        days.push(
            <button key={`day${currentDate.format('YMD')}`} className={classNames.join(' ')}>
                {currentDate.format('D')}
            </button>
        )
        currentDate.add(1, 'day')
    }

    return days
}

export default Month
