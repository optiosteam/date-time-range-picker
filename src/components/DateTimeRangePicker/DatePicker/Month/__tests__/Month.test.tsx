import React from 'react'
import { shallow } from 'enzyme'

import Month from '../Month'
import moment from 'moment'

describe('Date/time range picker month', () => {
    it('should render correctly', () => {
        const component = shallow(
            <Month
                month={5}
                year={2020}
                showDaysPreviousMonth={true}
                showDaysNextMonth={true}
                onDaySelected={() => undefined}
                onDayHover={() => undefined}
            />
        )

        expect(component).toMatchSnapshot()
    })

    it('should call selected callback', () => {
        const onDaySelectedMock = jest.fn()

        const component = shallow(
            <Month
                month={5}
                year={2020}
                showDaysPreviousMonth={true}
                showDaysNextMonth={true}
                onDaySelected={onDaySelectedMock}
                onDayHover={() => undefined}
            />
        )

        component.find('.date-time-range-picker-month-days button').at(10).simulate('click')
        expect(onDaySelectedMock).toBeCalledTimes(1)
        expect(onDaySelectedMock.mock.calls[0][0].format('YYYY-MM-DD')).toEqual('2020-05-06')
    })

    it('should call hover callback', () => {
        const onHoverMock = jest.fn()

        const component = shallow(
            <Month
                month={5}
                year={2020}
                showDaysPreviousMonth={true}
                showDaysNextMonth={true}
                onDaySelected={() => undefined}
                onDayHover={onHoverMock}
            />
        )

        component.find('.date-time-range-picker-month-days button').at(10).simulate('mouseenter')
        expect(onHoverMock).toBeCalledTimes(1)
        expect(onHoverMock.mock.calls[0][0].format('YYYY-MM-DD')).toEqual('2020-05-06')
    })

    it ('renders hover', () => {
        const component = shallow(
            <Month
                month={5}
                year={2020}
                showDaysPreviousMonth={true}
                showDaysNextMonth={true}
                fromDate={moment('2020-05-15')}
                hoverDate={moment('2020-05-10')}
                onDaySelected={() => undefined}
                onDayHover={() => undefined}
            />
        )

        expect(component).toMatchSnapshot()
    })

    it ('renders range', () => {
        const component = shallow(
            <Month
                month={5}
                year={2020}
                showDaysPreviousMonth={true}
                showDaysNextMonth={true}
                fromDate={moment('2020-05-10')}
                untilDate={moment('2020-05-15')}
                onDaySelected={() => undefined}
                onDayHover={() => undefined}
            />
        )

        expect(component).toMatchSnapshot()
    })
})
