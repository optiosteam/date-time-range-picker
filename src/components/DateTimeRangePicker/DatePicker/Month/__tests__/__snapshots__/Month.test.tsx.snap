// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date/time range picker month renders hover 1`] = `
<div
  className="date-time-range-picker-month"
>
  <div
    className="date-time-range-picker-month-daynames"
  >
    <div>
      Su
    </div>
    <div>
      Mo
    </div>
    <div>
      Tu
    </div>
    <div>
      We
    </div>
    <div>
      Th
    </div>
    <div>
      Fr
    </div>
    <div>
      Sa
    </div>
  </div>
  <div
    className="date-time-range-picker-month-days"
    onMouseLeave={[Function]}
  >
    <button
      className="other-month"
      key="day20200426"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      26
    </button>
    <button
      className="other-month"
      key="day20200427"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      27
    </button>
    <button
      className="other-month"
      key="day20200428"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      28
    </button>
    <button
      className="other-month"
      key="day20200429"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      29
    </button>
    <button
      className="other-month"
      key="day20200430"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      30
    </button>
    <button
      className=""
      key="day20200501"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      1
    </button>
    <button
      className=""
      key="day20200502"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      2
    </button>
    <button
      className=""
      key="day20200503"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      3
    </button>
    <button
      className=""
      key="day20200504"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      4
    </button>
    <button
      className=""
      key="day20200505"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      5
    </button>
    <button
      className=""
      key="day20200506"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      6
    </button>
    <button
      className=""
      key="day20200507"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      7
    </button>
    <button
      className=""
      key="day20200508"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      8
    </button>
    <button
      className=""
      key="day20200509"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      9
    </button>
    <button
      className="hover-range hover hover-past"
      key="day20200510"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      10
    </button>
    <button
      className="hover-range"
      key="day20200511"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      11
    </button>
    <button
      className="hover-range"
      key="day20200512"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      12
    </button>
    <button
      className="hover-range"
      key="day20200513"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      13
    </button>
    <button
      className="hover-range"
      key="day20200514"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      14
    </button>
    <button
      className="active active-from-date active-from-date-reverse hover-range"
      key="day20200515"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      15
    </button>
    <button
      className=""
      key="day20200516"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      16
    </button>
    <button
      className=""
      key="day20200517"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      17
    </button>
    <button
      className=""
      key="day20200518"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      18
    </button>
    <button
      className=""
      key="day20200519"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      19
    </button>
    <button
      className=""
      key="day20200520"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      20
    </button>
    <button
      className=""
      key="day20200521"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      21
    </button>
    <button
      className=""
      key="day20200522"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      22
    </button>
    <button
      className=""
      key="day20200523"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      23
    </button>
    <button
      className=""
      key="day20200524"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      24
    </button>
    <button
      className=""
      key="day20200525"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      25
    </button>
    <button
      className=""
      key="day20200526"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      26
    </button>
    <button
      className=""
      key="day20200527"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      27
    </button>
    <button
      className=""
      key="day20200528"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      28
    </button>
    <button
      className=""
      key="day20200529"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      29
    </button>
    <button
      className=""
      key="day20200530"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      30
    </button>
    <button
      className=""
      key="day20200531"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      31
    </button>
    <button
      className="other-month"
      key="day20200601"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      1
    </button>
    <button
      className="other-month"
      key="day20200602"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      2
    </button>
    <button
      className="other-month"
      key="day20200603"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      3
    </button>
    <button
      className="other-month"
      key="day20200604"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      4
    </button>
    <button
      className="other-month"
      key="day20200605"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      5
    </button>
    <button
      className="other-month"
      key="day20200606"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      6
    </button>
  </div>
</div>
`;

exports[`Date/time range picker month renders range 1`] = `
<div
  className="date-time-range-picker-month"
>
  <div
    className="date-time-range-picker-month-daynames"
  >
    <div>
      Su
    </div>
    <div>
      Mo
    </div>
    <div>
      Tu
    </div>
    <div>
      We
    </div>
    <div>
      Th
    </div>
    <div>
      Fr
    </div>
    <div>
      Sa
    </div>
  </div>
  <div
    className="date-time-range-picker-month-days"
    onMouseLeave={[Function]}
  >
    <button
      className="other-month"
      key="day20200426"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      26
    </button>
    <button
      className="other-month"
      key="day20200427"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      27
    </button>
    <button
      className="other-month"
      key="day20200428"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      28
    </button>
    <button
      className="other-month"
      key="day20200429"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      29
    </button>
    <button
      className="other-month"
      key="day20200430"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      30
    </button>
    <button
      className=""
      key="day20200501"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      1
    </button>
    <button
      className=""
      key="day20200502"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      2
    </button>
    <button
      className=""
      key="day20200503"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      3
    </button>
    <button
      className=""
      key="day20200504"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      4
    </button>
    <button
      className=""
      key="day20200505"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      5
    </button>
    <button
      className=""
      key="day20200506"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      6
    </button>
    <button
      className=""
      key="day20200507"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      7
    </button>
    <button
      className=""
      key="day20200508"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      8
    </button>
    <button
      className=""
      key="day20200509"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      9
    </button>
    <button
      className="active active-from-date in-range"
      key="day20200510"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      10
    </button>
    <button
      className="in-range"
      key="day20200511"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      11
    </button>
    <button
      className="in-range"
      key="day20200512"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      12
    </button>
    <button
      className="in-range"
      key="day20200513"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      13
    </button>
    <button
      className="in-range"
      key="day20200514"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      14
    </button>
    <button
      className="active active-until-date in-range"
      key="day20200515"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      15
    </button>
    <button
      className=""
      key="day20200516"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      16
    </button>
    <button
      className=""
      key="day20200517"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      17
    </button>
    <button
      className=""
      key="day20200518"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      18
    </button>
    <button
      className=""
      key="day20200519"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      19
    </button>
    <button
      className=""
      key="day20200520"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      20
    </button>
    <button
      className=""
      key="day20200521"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      21
    </button>
    <button
      className=""
      key="day20200522"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      22
    </button>
    <button
      className=""
      key="day20200523"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      23
    </button>
    <button
      className=""
      key="day20200524"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      24
    </button>
    <button
      className=""
      key="day20200525"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      25
    </button>
    <button
      className=""
      key="day20200526"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      26
    </button>
    <button
      className=""
      key="day20200527"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      27
    </button>
    <button
      className=""
      key="day20200528"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      28
    </button>
    <button
      className=""
      key="day20200529"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      29
    </button>
    <button
      className=""
      key="day20200530"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      30
    </button>
    <button
      className=""
      key="day20200531"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      31
    </button>
    <button
      className="other-month"
      key="day20200601"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      1
    </button>
    <button
      className="other-month"
      key="day20200602"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      2
    </button>
    <button
      className="other-month"
      key="day20200603"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      3
    </button>
    <button
      className="other-month"
      key="day20200604"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      4
    </button>
    <button
      className="other-month"
      key="day20200605"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      5
    </button>
    <button
      className="other-month"
      key="day20200606"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      6
    </button>
  </div>
</div>
`;

exports[`Date/time range picker month should render correctly 1`] = `
<div
  className="date-time-range-picker-month"
>
  <div
    className="date-time-range-picker-month-daynames"
  >
    <div>
      Su
    </div>
    <div>
      Mo
    </div>
    <div>
      Tu
    </div>
    <div>
      We
    </div>
    <div>
      Th
    </div>
    <div>
      Fr
    </div>
    <div>
      Sa
    </div>
  </div>
  <div
    className="date-time-range-picker-month-days"
    onMouseLeave={[Function]}
  >
    <button
      className="other-month"
      key="day20200426"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      26
    </button>
    <button
      className="other-month"
      key="day20200427"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      27
    </button>
    <button
      className="other-month"
      key="day20200428"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      28
    </button>
    <button
      className="other-month"
      key="day20200429"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      29
    </button>
    <button
      className="other-month"
      key="day20200430"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      30
    </button>
    <button
      className=""
      key="day20200501"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      1
    </button>
    <button
      className=""
      key="day20200502"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      2
    </button>
    <button
      className=""
      key="day20200503"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      3
    </button>
    <button
      className=""
      key="day20200504"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      4
    </button>
    <button
      className=""
      key="day20200505"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      5
    </button>
    <button
      className=""
      key="day20200506"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      6
    </button>
    <button
      className=""
      key="day20200507"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      7
    </button>
    <button
      className=""
      key="day20200508"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      8
    </button>
    <button
      className=""
      key="day20200509"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      9
    </button>
    <button
      className=""
      key="day20200510"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      10
    </button>
    <button
      className=""
      key="day20200511"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      11
    </button>
    <button
      className=""
      key="day20200512"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      12
    </button>
    <button
      className=""
      key="day20200513"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      13
    </button>
    <button
      className=""
      key="day20200514"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      14
    </button>
    <button
      className=""
      key="day20200515"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      15
    </button>
    <button
      className=""
      key="day20200516"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      16
    </button>
    <button
      className=""
      key="day20200517"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      17
    </button>
    <button
      className=""
      key="day20200518"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      18
    </button>
    <button
      className=""
      key="day20200519"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      19
    </button>
    <button
      className=""
      key="day20200520"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      20
    </button>
    <button
      className=""
      key="day20200521"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      21
    </button>
    <button
      className=""
      key="day20200522"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      22
    </button>
    <button
      className=""
      key="day20200523"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      23
    </button>
    <button
      className=""
      key="day20200524"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      24
    </button>
    <button
      className=""
      key="day20200525"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      25
    </button>
    <button
      className=""
      key="day20200526"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      26
    </button>
    <button
      className=""
      key="day20200527"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      27
    </button>
    <button
      className=""
      key="day20200528"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      28
    </button>
    <button
      className=""
      key="day20200529"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      29
    </button>
    <button
      className=""
      key="day20200530"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      30
    </button>
    <button
      className=""
      key="day20200531"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      31
    </button>
    <button
      className="other-month"
      key="day20200601"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      1
    </button>
    <button
      className="other-month"
      key="day20200602"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      2
    </button>
    <button
      className="other-month"
      key="day20200603"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      3
    </button>
    <button
      className="other-month"
      key="day20200604"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      4
    </button>
    <button
      className="other-month"
      key="day20200605"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      5
    </button>
    <button
      className="other-month"
      key="day20200606"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      6
    </button>
  </div>
</div>
`;
